packages/calypso-e2e/src/lib/blocks/image-block.ts

```diff
@@ -63,6 +72,40 @@ export class ImageBlock {
 		return await this.getImage();
 	}
 
+	/**
+	 * Given path to an image file, uploads the image to an Image block
+	 * using the media modal via `Select Images` > `Media Library`.
+	 *
+	 * @param {string} path Path to the image file.
+	 */
+	async uploadFromModal( path: string ): Promise< ElementHandle > {
+		const frame = ( await this.block.ownerFrame() ) as Frame;
+		const page = frame.page();
+
+		// The resulting React modal is essentially the MediaPage, just
+		// embedded within the modal, with additional buttons to
+		// confirm or cancel image insertion.
+		const mediaPage = new MediaPage( page );
+		await mediaPage.upload( path );
+		await mediaPage.clickModalButton( 'Confirm' );
+		await this.waitUntilUploaded();
+		return await this.getImage();
+	}
+
+	/**
+	 * Click on the `Select Image` button, visible when no image is selected.
+	 *
+	 * @param {Sources} source Source for the image.
+	 */
+	async selectImageSource( source: Sources ): Promise< void > {
+		const buttonHandle = await this.block.waitForSelector( selectors.selectImageSourceButton );
+		await buttonHandle.click();
```

Comment:
Very small nit, but since you're using the `locator` API in the next couple lines, should we use that here too? It also tends to be more stable than using element handles.

Reply:
Good call, thanks!

Reply:
Okay, so looks like it is not possible to use the `locator` call from an ElementHandle, which is what `this.block` boils down to.

Reply:
Ahhh right, good point! All good 👍